lib/bioinfo/blast.ml

```diff
@@ -0,0 +1,72 @@
+open Core
+open Bistro.Std
+open Bistro.EDSL
+open Bistro_bioinfo.Std
+
+type db = [`blast_db] directory
+let env = docker_image ~account:"pveber" ~name:"ncbi-blast" ~tag:"2.4.0" ()
+
+let db_name = "db"
+
+let fastadb fa dbtype = 
+	workflow ~descr:"blast.makedb" [
+		mkdir_p dest ; 
+		cmd ~env "makeblastdb" [
+			opt "-in" dep fa ;
+			opt "-dbtype" ident dbtype ; 
```

Comment:
the type of `dbtype` is `Bistro.Template.t`, which is really not convenient for a user. I'd much prefer have a polymorphic variant type here `[`nucl | `prot]`

Reply:
So we have to create the type ? In the .ml ? I don't really see how to call the function then. 
Something like `type polymorphic_variant = Prot | Nucl` in .ml and call `Blast.makedb ~dbtype:Prot ref_prot ` outside ? With a "reconversion" to string somewhere to call the command line.
(it's not exactly like that because it's not working)

Reply:
actually if you use polymorphic variants `` `Nucl`` and `` `Prot``, you don't need to declare the type and it's also easier when calling the function. So yes, basically a call would be just like you say ``Blast.makedb ~dbtype:`Prot ref_prot``, and yes you have to provide the conversion function that will take a variant and make an appropriate `Bistro.Template.t` out of it.